client/state/jetpack-connect/reducer/test/jetpack-auth-attempts.js

```diff
@@ -2,52 +2,59 @@
 /**
  * Internal dependencies
  */
-import jetpackAuthAttempts from '../jetpack-auth-attempts';
-import { JETPACK_CONNECT_RETRY_AUTH } from 'state/action-types';
-
-describe( '#jetpackAuthAttempts()', () => {
-	test( 'should default to an empty object', () => {
-		const state = jetpackAuthAttempts( undefined, {} );
-		expect( state ).toEqual( {} );
-	} );
+import { authAttempts } from '../jetpack-auth-attempts';
+import { JETPACK_CONNECT_COMPLETE_FLOW, JETPACK_CONNECT_RETRY_AUTH } from 'state/action-types';
 
+describe( '#authAttempts()', () => {
 	test( 'should update the timestamp when adding an existent slug with stale timestamp', () => {
-		const state = jetpackAuthAttempts(
-			{ 'example.com': { timestamp: 1, attempt: 1 } },
+		const state = authAttempts(
+			{ timestamp: 1, attempt: 1 },
 			{
 				type: JETPACK_CONNECT_RETRY_AUTH,
 				slug: 'example.com',
 				attemptNumber: 2,
 			}
 		);
-		expect( state[ 'example.com' ] ).toMatchObject( {
+		expect( state ).toMatchObject( {
 			timestamp: expect.any( Number ),
 		} );
 	} );
 
 	test( 'should reset the attempt number to 0 when adding an existent slug with stale timestamp', () => {
-		const state = jetpackAuthAttempts(
-			{ 'example.com': { timestamp: 1, attempt: 1 } },
+		const state = authAttempts(
+			{ timestamp: 1, attempt: 1 },
 			{
 				type: JETPACK_CONNECT_RETRY_AUTH,
 				slug: 'example.com',
 				attemptNumber: 2,
 			}
 		);
 
-		expect( state[ 'example.com' ] ).toMatchObject( { attempt: 0 } );
+		expect( state ).toMatchObject( { attempt: 0 } );
 	} );
 
 	test( 'should store the attempt number when adding an existent slug with non-stale timestamp', () => {
-		const state = jetpackAuthAttempts(
-			{ 'example.com': { timestamp: Date.now(), attempt: 1 } },
+		const state = authAttempts(
+			{ timestamp: Date.now(), attempt: 1 },
 			{
 				type: JETPACK_CONNECT_RETRY_AUTH,
 				slug: 'example.com',
 				attemptNumber: 2,
 			}
 		);
 
-		expect( state[ 'example.com' ] ).toMatchObject( { attempt: 2 } );
+		expect( state ).toMatchObject( { attempt: 2 } );
+	} );
+
+	test( 'should clear state on completion', () => {
+		const state = authAttempts(
+			{ timestamp: Date.now(), attempt: 1 },
+			{
+				type: JETPACK_CONNECT_COMPLETE_FLOW,
+				slug: 'example.com',
+			}
+		);
+
+		expect( state ).not.toBeDefined();
```

Comment:
Tests above only test the `authAttempts` inner reducer; I'd also add a test for the default reducer that we export, in order to verify proper usage of the keyed reducer there. Right now tests don't care whether we're keying by anything in the default reducer. This is also confirmed by the fact that this inner reducer will not care at all about the `slug` that we pass in the action types - and this is probably something we need to test in the keyed reducer tests.

Reply:
Added a reducer test in f57e2a0